pdns/dnsmessage.proto

```diff
@@ -126,6 +126,36 @@ message PBDNSMessage {
     required MetaValue value = 2;
   }
   repeated Meta meta = 22;                      // Arbitrary meta-data - to be used in future rather than adding new fields all the time
+
+  // The well known EventTrace event numbers
+  enum EventType {
+    CustomEvent = 0;                            // A custom event
+    RecRecv = 1;                                // A request was received by the recursor process
+    PCacheCheck = 2;                            // A packet cache check was initiated or completed; value: bool cacheHit
+    SyncRes = 3;                                // Syncres main function has started or completed; value: int rcode
+    AnswerSent = 4;                             // The answer was sent to the client
+    LuaGetTag = 100;                            // Events below mark start or end of Lua hook calls; value: return value of hook
+    LuaGetTagFFI = 101;
+    LuaIPFilter = 102;
+    LuaPreRPZ = 103;
+    LuaPreResolve = 104;
+    LuaPreOutQuery = 105;
+    LuaPostResolve = 106;
+    LuaNoData = 107;
+    LuaNXDomain = 108;
+  }
+
+  message Event {
+    required int64 ts = 1;                      // Timestamp in ns relative to time of creation of event trace data structure
```

Comment:
I'm not sure I see a case where this can be negative? If we know some events happen before the creation of the event trace structure, perhaps it would make sense to set the base time of event trace structure to the first of these events when it is created, provided that it can be done?

Reply:
An example of events with a negative (relative) timestamps are kernel provided packet timestamps. By defintion they are eaelri than the eventTrace data strcuture creation (the current base ts). If we change the base beacuse of such an eaelier timestamp, all already added event timestamps should be adapted as well.

Reply:
Understood, ans I guess it's not easy to pass that timestamp at the creation of the eventTrace data structure. That means the receiver will either need to work with negative time offsets as well or scan the events an compute the relative timestamps, I guess?

Reply:
The timestamps are already stored relatively. Alternatively I could do a "normalize" step when exporting, so that the oldest timestamp becomes zero, but it will cost some effort. I'll think about it.

Reply:
`int64` makes calculating differences safer. The same proto will also be used by consumers that will base their type on this definition.

Ideally we would avoid actually sending negative timestamp. If we do want to send those, perhaps use `sint64`, which uses a zig-zag encoding to prevent negative numbers from using the maximum size in the wire protocol.

Reply:
I decided to keep the int64_t. When a event is added before the current `d_base`, its timestamp wil become the new `d_base` and existing events will be adjusted to be relative to the new `d_base`.

The main reason I'm using relative timestamps is hat it will generate much smaller values, so that the protobuf encoding will be relatively small as well.

I also discovered that kernel packet timestamps use an unspecified time base, so it might be hard to fit them in. But that is for the future.